src/libsync/vfs/cfapi/vfs_cfapi.cpp

```diff
@@ -396,8 +396,19 @@ void VfsCfApi::requestHydration(const QString &requestId, const QString &path)
         return;
     }
 
+    bool isNotVirtualFileFailure = false;
     if (!record.isVirtualFile()) {
-        qCInfo(lcCfApi) << "Couldn't hydrate, the file is not virtual";
+        if (isDehydratedPlaceholder(path)) {
+            qCWarning(lcCfApi) << "Hydration requested for a placeholder file not marked as virtual in local DB. Attempting to fix it...";
+            record._type = ItemTypeVirtualFileDownload;
+            isNotVirtualFileFailure = !journal->setFileRecord(record);
+        } else {
+            isNotVirtualFileFailure = true;
+        }
+    }
+
+    if (isNotVirtualFileFailure) {
+        qCWarning(lcCfApi) << "Couldn't hydrate, the file is not virtual";
```

Comment:
seems to me something I could reproduce
like doing those steps:

1. implicitly hydrate a very big file
2. override it with a real file during the hydration
3. check that you get an error

Reply:
@mgallien 
yes, but, then I am able to hydrate that file again after closing the popup or restarting the desktop client, so not the same as this issue, the file is not hydrated in this case but it is online-only placeholder